### PR TITLE
Cleanup another round of Coverity issues

### PR DESCRIPTION
**Pull request by kcgen**

### CODE REVIEW

File: src/libs/nuked/opl3.c [hunk 1]
```diff
@@ -1280,6 +1281,7 @@ void OPL3_Reset(opl3_chip *chip, uint32_t samplerate)
         chip->slot[local_ch_slot + 3].channel = channel;
         if ((channum % 9) < 3)
         {
+            assert(channum <= 11);
```
Comment (johnnovak) [inline src/libs/nuked/opl3.c:1284]: I'm not quite sure where 11 is coming from.

Comment (kcgen) [reply src/libs/nuked/opl3.c:1284]: Will replace it with a constexpr calculation. (Unfortunately the baseline code doesn't define constants.. they use magic values everywhere, but I'll drop a couple in).

File: src/misc/support.cpp [hunk 2]
```diff
@@ -246,24 +246,28 @@ static const std::deque<std_fs::path> &GetResourceParentPaths()
...
+		return distribute(generator);
 	};
 }
 // Explicit template instantiations
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-template std::function<char()> CreateRandomizer<char>(const char, const char);
-template std::function<uint8_t()> CreateRandomizer<uint8_t>(const uint8_t,
-                                                            const uint8_t);
+template std::function<int16_t()> CreateRandomizer<int16_t>(const int16_t, const int16_t);
+template std::function<float()> CreateRandomizer<float>(const float, const float);
```
Comment (johnnovak) [inline src/misc/support.cpp:270]: Noice 😎

File: tests/support_tests.cpp [hunk 3]
```diff
@@ -156,4 +156,88 @@ TEST(Support_next_uint, UnsignedInvalid)
 	EXPECT_NE(typeid(next_uint_t<uint64_t>), typeid(uint32_t));
 }
 
+template <typename T>
+void test_randomizer(const T min_value, const T max_value)
+{
+	// Ensure the range is valid
+	ASSERT_LT(min_value, max_value);
+
+	// Calculate a quarter of the range; we'll this to roughly test the distribution of the genrated values
```
Comment (johnnovak) [inline tests/support_tests.cpp:165]: typos: "we'll this", "genrated"

File: tests/support_tests.cpp [hunk 4]
```diff
@@ -156,4 +156,88 @@ TEST(Support_next_uint, UnsignedInvalid)
...
+	// positive percent-as-ratio
+	test_randomizer(0.0f, 1.0f);
+
+	// negative percent-as-ratio
+	test_randomizer(-1.0f, 0.0f);
+
+	// postitive and negative percent-as-ratio
+	test_randomizer(-1.0f, 1.0f);
+}
+
```
Comment (johnnovak) [inline tests/support_tests.cpp:242]: Good to see the unit test coverage increasing for our utility functions 💯

**Review by johnnovak — APPROVED** [review]
Looking good @kcgen 👍🏻

**Comment by kcgen** [comment]
Thanks for the review @johnnovak !
Will get these fixes in over the next day.

**Comment by kcgen** [comment]
Thanks for the review, @johnnovak.
Will get this merged and see what the scan shows.

**Pull request merged**